src/patito/_pydantic/dtypes/utils.py

```diff
@@ -93,7 +96,7 @@ def is_optional(type_annotation: type[Any] | Any | None) -> bool:
 
 def parse_composite_dtype(dtype: DataTypeClass | DataType) -> str:
     """For serialization, converts polars dtype to string representation."""
-    if dtype in pl.NESTED_DTYPES:
+    if dtype in NESTED_DTYPES:
```

Comment:
If you know `dtype` is indeed a `DataTypeClass | DataType`, you can use `is_nested`:

```suggestion
    if dtype.is_nested():
```